app/controllers/jobseekers/job_applications_controller.rb

```diff
@@ -144,19 +151,7 @@ def update_jobseeker_profile!(job_application)
   def all_steps_valid?
     # Check that all steps are valid, in case we have changed the validations since the step was completed.
     # NB: Only validates top-level step forms. Does not validate individual qualifications, employments, or references.
-    step_process.steps.excluding(:review).all? { |step| step_valid?(step) }
-  end
-
-  def step_valid?(step)
-    form_class = "jobseekers/job_application/#{step}_form".camelize.constantize
-
-    attributes = form_class.load_form(job_application)
-
-    form = form_class.new(attributes)
-
-    form.valid?.tap do
-      job_application.errors.merge!(form.errors)
-    end
+    Jobseekers::JobApplications::JobApplicationHandler.new(job_application, step_process).all_steps_valid?
```

Comment:
did you consider just having a different step_process class for the new application type?